x-pack/plugins/security_solution/public/flyout/preview/components/rule_preview_title.tsx

```diff
@@ -0,0 +1,131 @@
+
+interface RulePreviewTitleProps {
+  /**
+   * Rule object that represents relevant information about a rule
+   */
+  rule: Rule;
+  /**
+   * Tooltip text that explains why a user does not have permission to rule
+   */
+  tooltipText?: string;
+  /**
+   * Boolean that indivates whether the rule switch button is isabled
+   */
+  isButtonDisabled: boolean;
+  /**
+   * Boolean that indivates whether the rule switch shoud be enabled
+   */
+  isRuleEnabled: boolean;
+}
+
+/**
+ * Title component that shows basic information of a rule. This is displayed above rule preview body in rule preview panel
+ */
+export const RulePreviewTitle: React.FC<RulePreviewTitleProps> = ({
+  rule,
+  tooltipText,
+  isButtonDisabled,
+  isRuleEnabled,
+}) => {
+  const { startMlJobs } = useStartMlJobs();
+  const startMlJobsIfNeeded = useCallback(
+    () => startMlJobs(rule?.machine_learning_job_id),
+    [rule, startMlJobs]
+  );
+
+  const createdBy = useMemo(
+    () => (
+      <EuiText size="xs">
+        <FormattedMessage
+          id="xpack.securitySolution.flyout.documentDetails.ruleCreationDescription"
+          defaultMessage="Created by: {by} on {date}"
+          values={{
+            by: rule?.created_by ?? i18n.UNKNOWN_TEXT,
+            date: (
+              <FormattedDate
+                value={rule?.created_at ?? new Date().toISOString()}
+                fieldName="createdAt"
+              />
+            ),
+          }}
+        />
+      </EuiText>
+    ),
+    [rule]
+  );
+
+  const updatedBy = useMemo(
+    () => (
+      <EuiText size="xs">
+        <FormattedMessage
+          id="xpack.securitySolution.flyout.documentDetails.ruleUpdateDescription"
+          defaultMessage="Updated by: {by} on {date}"
+          values={{
+            by: rule?.updated_by ?? i18n.UNKNOWN_TEXT,
+            date: (
+              <FormattedDate
+                value={rule?.updated_at ?? new Date().toISOString()}
+                fieldName="updatedAt"
+              />
+            ),
+          }}
+        />
+      </EuiText>
+    ),
+    [rule]
+  );
+
+  const enableRule = useMemo(
+    () => (
+      <EuiToolTip position="top" content={tooltipText}>
+        <EuiFlexGroup gutterSize="s" alignItems="center">
+          <EuiFlexItem grow={false}>
+            <RuleSwitch
+              id={rule?.id ?? '-1'}
+              isDisabled={isButtonDisabled}
+              enabled={isRuleEnabled}
+              startMlJobsIfNeeded={startMlJobsIfNeeded}
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>{i18n.ENABLE_RULE_TEXT}</EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiToolTip>
+    ),
+    [rule, tooltipText, isButtonDisabled, isRuleEnabled, startMlJobsIfNeeded]
+  );
+
+  return (
+    <div data-test-subj={RULE_PREVIEW_TITLE_TEST_ID}>
+      <EuiTitle>
+        <h6>{rule.name}</h6>
+      </EuiTitle>
+      <EuiSpacer size="s" />
+      <EuiFlexGroup gutterSize="xs" direction="column">
```

Comment:
what do you think of this slightly different layout?
```
<EuiFlexGroup gutterSize="xs" alignItems="center">
  <EuiFlexItem>
    <EuiFlexGroup gutterSize="xs" direction="column">
      <EuiFlexItem data-test-subj={RULE_PREVIEW_RULE_CREATED_BY_TEST_ID}>
        {createdBy}
      </EuiFlexItem>
      <EuiFlexItem data-test-subj={RULE_PREVIEW_RULE_UPDATED_BY_TEST_ID}>
        {updatedBy}
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiFlexItem>
  <EuiFlexItem grow={false}>{enableRule}</EuiFlexItem>
</EuiFlexGroup>
```

It would look like this 
![Screenshot 2023-08-04 at 6 01 05 PM](https://github.com/elastic/kibana/assets/17276605/2a05e1d0-dc1b-4ba8-b798-2277095445ca)

Reply:
after discussion let's not worry about this UI for now